src/v/cluster/controller_backend.cc

```diff
@@ -1758,7 +1767,11 @@ ss::future<std::error_code> controller_backend::transfer_partition(
 
     auto partition = _partition_manager.local().get(ntp);
     if (partition) {
-        co_await shutdown_partition(std::move(partition));
+        auto xst_state = co_await shutdown_partition(std::move(partition));
+        co_await container().invoke_on(
+          destination, [&ntp, &xst_state](controller_backend& dest) {
+              dest._xst_states[ntp] = xst_state;
```

Comment:
This will be a cross-shard move (in c++ sense), which means that the object has to be freed on a shard that is different from where it was allocated. While this will work, this is discouraged and may lead to subtle bugs (if there is a custom destructor that is not expecting this). So we have to either copy or use `ss::foreign_ptr` here, and the latter seems like a bit of an overkill for now :)